apps/server/src/modules/account/uc/account.uc.ts

```diff
@@ -285,9 +282,7 @@ export class AccountUc {
 			throw new ForbiddenOperationError('External account details can not be changed.');
 		}
 
-		if (!account.password) {
-			throw new Error('The account does not have a password to compare against.');
-		} else if (await this.checkPassword(password, account.password)) {
+		if (await this.accountService.validatePassword(account, password)) {
```

Comment:
If the password is valid, an Error is thrown?

Reply:
The UC wants to change the current password. This checks if the new password matches the old password. If so, the password was not changed but the previous password was taken again. So matching password is indeed an error case here.

Reply:
Can we put this check into it's own private function to make it's function more clear?

Reply:
Is the thrown error not clear enough?